diff --git a/src/amd/vulkan/radv_formats.c b/src/amd/vulkan/radv_formats.c
index a839e2db43d..bf76b6b2551 100644
--- a/src/amd/vulkan/radv_formats.c
+++ b/src/amd/vulkan/radv_formats.c
@@ -29,7 +29,7 @@
 
 #include "vk_util.h"
 
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/format_srgb.h"
 #include "util/format_r11g11b10f.h"
 #include "util/format_rgb9e5.h"
diff --git a/src/freedreno/fdl/fd5_layout_test.c b/src/freedreno/fdl/fd5_layout_test.c
index 71ed259e1fc..e8496996d3b 100644
--- a/src/freedreno/fdl/fd5_layout_test.c
+++ b/src/freedreno/fdl/fd5_layout_test.c
@@ -24,7 +24,8 @@
 #include "freedreno_layout.h"
 #include "fd_layout_test.h"
 #include "adreno_common.xml.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
+#include "util/u_math.h"
 #include "a5xx.xml.h"
 
 #include <stdio.h>
diff --git a/src/freedreno/perfcntrs/fd2_perfcntr.c b/src/freedreno/perfcntrs/fd2_perfcntr.c
index eac2de0214b..a0f1ef8b3c0 100644
--- a/src/freedreno/perfcntrs/fd2_perfcntr.c
+++ b/src/freedreno/perfcntrs/fd2_perfcntr.c
@@ -25,7 +25,8 @@
  *    Rob Clark <robclark@freedesktop.org>
  */
 
-#include "util/u_half.h"
+#include "util/half_float.h"
+#include "util/u_math.h"
 #include "adreno_common.xml.h"
 #include "adreno_pm4.xml.h"
 #include "a2xx.xml.h"
diff --git a/src/freedreno/perfcntrs/fd5_perfcntr.c b/src/freedreno/perfcntrs/fd5_perfcntr.c
index 2d0579ca363..0d8d2ae47b5 100644
--- a/src/freedreno/perfcntrs/fd5_perfcntr.c
+++ b/src/freedreno/perfcntrs/fd5_perfcntr.c
@@ -27,7 +27,8 @@
 #ifndef FD5_PERFCNTR_H_
 #define FD5_PERFCNTR_H_
 
-#include "util/u_half.h"
+#include "util/half_float.h"
+#include "util/u_math.h"
 #include "adreno_common.xml.h"
 #include "a5xx.xml.h"
 
diff --git a/src/freedreno/perfcntrs/fd6_perfcntr.c b/src/freedreno/perfcntrs/fd6_perfcntr.c
index 02e55f8babd..f50a72e3bac 100644
--- a/src/freedreno/perfcntrs/fd6_perfcntr.c
+++ b/src/freedreno/perfcntrs/fd6_perfcntr.c
@@ -27,7 +27,8 @@
 #ifndef FD6_PERFCNTR_H_
 #define FD6_PERFCNTR_H_
 
-#include "util/u_half.h"
+#include "util/half_float.h"
+#include "util/u_math.h"
 #include "adreno_common.xml.h"
 #include "adreno_pm4.xml.h"
 #include "a6xx.xml.h"
diff --git a/src/freedreno/perfcntrs/freedreno_perfcntr.h b/src/freedreno/perfcntrs/freedreno_perfcntr.h
index 867048afc1c..250ea92fc37 100644
--- a/src/freedreno/perfcntrs/freedreno_perfcntr.h
+++ b/src/freedreno/perfcntrs/freedreno_perfcntr.h
@@ -27,6 +27,8 @@
 #ifndef FREEDRENO_PERFCNTR_H_
 #define FREEDRENO_PERFCNTR_H_
 
+#include "util/macros.h"
+
 /*
  * Mapping very closely to the AMD_performance_monitor extension, adreno has
  * groups of performance counters where each group has N counters, which can
diff --git a/src/freedreno/vulkan/tu_clear_blit.c b/src/freedreno/vulkan/tu_clear_blit.c
index d2f112c6fd4..7c849962f49 100644
--- a/src/freedreno/vulkan/tu_clear_blit.c
+++ b/src/freedreno/vulkan/tu_clear_blit.c
@@ -14,7 +14,7 @@
 #include "util/format_r11g11b10f.h"
 #include "util/format_rgb9e5.h"
 #include "util/format_srgb.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 static uint32_t
 tu_pack_float32_for_unorm(float val, int bits)
diff --git a/src/gallium/auxiliary/Makefile.sources b/src/gallium/auxiliary/Makefile.sources
index e66548876b7..f8ef3826188 100644
--- a/src/gallium/auxiliary/Makefile.sources
+++ b/src/gallium/auxiliary/Makefile.sources
@@ -248,7 +248,6 @@ C_SOURCES := \
 	util/u_framebuffer.h \
 	util/u_gen_mipmap.c \
 	util/u_gen_mipmap.h \
-	util/u_half.h \
 	util/u_handle_table.c \
 	util/u_handle_table.h \
 	util/u_hash_table.c \
diff --git a/src/gallium/auxiliary/gallivm/lp_bld_const.c b/src/gallium/auxiliary/gallivm/lp_bld_const.c
index 63a6599284f..18ece7324d7 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_const.c
+++ b/src/gallium/auxiliary/gallivm/lp_bld_const.c
@@ -37,7 +37,7 @@
 
 #include "util/u_debug.h"
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 #include "lp_bld_type.h"
 #include "lp_bld_const.h"
diff --git a/src/gallium/auxiliary/gallivm/lp_bld_conv.c b/src/gallium/auxiliary/gallivm/lp_bld_conv.c
index 9edc1614f38..c68b8850473 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_conv.c
+++ b/src/gallium/auxiliary/gallivm/lp_bld_conv.c
@@ -63,7 +63,7 @@
 
 #include "util/u_debug.h"
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_cpu_detect.h"
 
 #include "lp_bld_type.h"
diff --git a/src/gallium/auxiliary/meson.build b/src/gallium/auxiliary/meson.build
index 72f569ac832..37a8e011acd 100644
--- a/src/gallium/auxiliary/meson.build
+++ b/src/gallium/auxiliary/meson.build
@@ -264,7 +264,6 @@ files_libgallium = files(
   'util/u_framebuffer.h',
   'util/u_gen_mipmap.c',
   'util/u_gen_mipmap.h',
-  'util/u_half.h',
   'util/u_handle_table.c',
   'util/u_handle_table.h',
   'util/u_hash_table.c',
diff --git a/src/gallium/auxiliary/tgsi/tgsi_exec.c b/src/gallium/auxiliary/tgsi/tgsi_exec.c
index 95c2ec01822..4a009f7a59a 100644
--- a/src/gallium/auxiliary/tgsi/tgsi_exec.c
+++ b/src/gallium/auxiliary/tgsi/tgsi_exec.c
@@ -58,7 +58,7 @@
 #include "tgsi/tgsi_parse.h"
 #include "tgsi/tgsi_util.h"
 #include "tgsi_exec.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_memory.h"
 #include "util/u_math.h"
 #include "util/rounding.h"
diff --git a/src/gallium/auxiliary/translate/translate_generic.c b/src/gallium/auxiliary/translate/translate_generic.c
index 7af2f93a46a..3f07a58fc84 100644
--- a/src/gallium/auxiliary/translate/translate_generic.c
+++ b/src/gallium/auxiliary/translate/translate_generic.c
@@ -32,7 +32,7 @@
 
 #include "util/u_memory.h"
 #include "util/format/u_format.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_math.h"
 #include "pipe/p_state.h"
 #include "translate.h"
diff --git a/src/gallium/auxiliary/util/u_half.h b/src/gallium/auxiliary/util/u_half.h
deleted file mode 100644
index bb1a048b2c2..00000000000
--- a/src/gallium/auxiliary/util/u_half.h
+++ /dev/null
@@ -1,36 +0,0 @@
-/**************************************************************************
- *
- * Copyright 2010 Luca Barbieri
- *
- * Permission is hereby granted, free of charge, to any person obtaining
- * a copy of this software and associated documentation files (the
- * "Software"), to deal in the Software without restriction, including
- * without limitation the rights to use, copy, modify, merge, publish,
- * distribute, sublicense, and/or sell copies of the Software, and to
- * permit persons to whom the Software is furnished to do so, subject to
- * the following conditions:
- *
- * The above copyright notice and this permission notice (including the
- * next paragraph) shall be included in all copies or substantial
- * portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
- * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
- * IN NO EVENT SHALL THE COPYRIGHT OWNER(S) AND/OR ITS SUPPLIERS BE
- * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
- * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
- * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
- *
- **************************************************************************/
-
-
-#ifndef U_HALF_H
-#define U_HALF_H
-
-#include "pipe/p_compiler.h"
-#include "util/u_math.h"
-#include "util/half_float.h"
-
-#endif /* U_HALF_H */
-
diff --git a/src/gallium/drivers/etnaviv/etnaviv_blend.c b/src/gallium/drivers/etnaviv/etnaviv_blend.c
index b0479609b9c..36624265654 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_blend.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_blend.c
@@ -32,7 +32,7 @@
 #include "hw/common.xml.h"
 #include "pipe/p_defines.h"
 #include "util/u_memory.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 void *
 etna_blend_state_create(struct pipe_context *pctx,
diff --git a/src/gallium/drivers/etnaviv/etnaviv_compiler_nir.c b/src/gallium/drivers/etnaviv/etnaviv_compiler_nir.c
index 8b3296a9f9e..27aa2c21379 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_compiler_nir.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_compiler_nir.c
@@ -42,7 +42,7 @@
 #include "compiler/nir/nir_builder.h"
 
 #include "tgsi/tgsi_strings.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 static bool
 etna_alu_to_scalar_filter_cb(const nir_instr *instr, const void *data)
diff --git a/src/gallium/drivers/etnaviv/etnaviv_disasm.c b/src/gallium/drivers/etnaviv/etnaviv_disasm.c
index 0be6d9e5a37..67eb400e074 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_disasm.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_disasm.c
@@ -34,7 +34,7 @@
 
 #include "hw/isa.xml.h"
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 struct instr {
    /* dword0: */
diff --git a/src/gallium/drivers/etnaviv/etnaviv_zsa.c b/src/gallium/drivers/etnaviv/etnaviv_zsa.c
index 8e9ced460d6..2f9f2cb13c1 100644
--- a/src/gallium/drivers/etnaviv/etnaviv_zsa.c
+++ b/src/gallium/drivers/etnaviv/etnaviv_zsa.c
@@ -29,7 +29,7 @@
 #include "etnaviv_context.h"
 #include "etnaviv_screen.h"
 #include "etnaviv_translate.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_memory.h"
 
 #include "hw/common.xml.h"
diff --git a/src/gallium/drivers/freedreno/freedreno_util.h b/src/gallium/drivers/freedreno/freedreno_util.h
index 248d6552e78..d17d80dc816 100644
--- a/src/gallium/drivers/freedreno/freedreno_util.h
+++ b/src/gallium/drivers/freedreno/freedreno_util.h
@@ -34,7 +34,7 @@
 #include "pipe/p_state.h"
 #include "util/u_debug.h"
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_dynarray.h"
 #include "util/u_pack_color.h"
 
diff --git a/src/gallium/drivers/lima/ir/pp/codegen.c b/src/gallium/drivers/lima/ir/pp/codegen.c
index 8b689cbdf41..11d4e1a3857 100644
--- a/src/gallium/drivers/lima/ir/pp/codegen.c
+++ b/src/gallium/drivers/lima/ir/pp/codegen.c
@@ -23,7 +23,7 @@
  */
 
 #include "util/ralloc.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/bitscan.h"
 
 #include "ppir.h"
diff --git a/src/gallium/drivers/lima/ir/pp/disasm.c b/src/gallium/drivers/lima/ir/pp/disasm.c
index 3ea500b9512..50aa4cbb852 100644
--- a/src/gallium/drivers/lima/ir/pp/disasm.c
+++ b/src/gallium/drivers/lima/ir/pp/disasm.c
@@ -24,7 +24,7 @@
  *
  */
 
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 #include "ppir.h"
 #include "codegen.h"
diff --git a/src/gallium/drivers/lima/lima_draw.c b/src/gallium/drivers/lima/lima_draw.c
index 672108400a7..5c60eee2c70 100644
--- a/src/gallium/drivers/lima/lima_draw.c
+++ b/src/gallium/drivers/lima/lima_draw.c
@@ -25,7 +25,7 @@
 
 #include "util/format/u_format.h"
 #include "util/u_debug.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_helpers.h"
 #include "util/u_inlines.h"
 #include "util/u_pack_color.h"
diff --git a/src/gallium/drivers/nouveau/nv30/nv30_state_validate.c b/src/gallium/drivers/nouveau/nv30/nv30_state_validate.c
index 9fbf9b37cee..1276561895f 100644
--- a/src/gallium/drivers/nouveau/nv30/nv30_state_validate.c
+++ b/src/gallium/drivers/nouveau/nv30/nv30_state_validate.c
@@ -25,7 +25,7 @@
 
 #include "util/format/u_format.h"
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 #include "nv_object.xml.h"
 #include "nv30/nv30-40_3d.xml.h"
diff --git a/src/gallium/drivers/r300/r300_blit.c b/src/gallium/drivers/r300/r300_blit.c
index b627e013898..8d3ba1381d5 100644
--- a/src/gallium/drivers/r300/r300_blit.c
+++ b/src/gallium/drivers/r300/r300_blit.c
@@ -26,7 +26,7 @@
 #include "r300_reg.h"
 
 #include "util/format/u_format.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_pack_color.h"
 #include "util/u_surface.h"
 
diff --git a/src/gallium/drivers/r300/r300_state.c b/src/gallium/drivers/r300/r300_state.c
index 8d3db508a33..7f120f00aff 100644
--- a/src/gallium/drivers/r300/r300_state.c
+++ b/src/gallium/drivers/r300/r300_state.c
@@ -24,7 +24,7 @@
 #include "draw/draw_context.h"
 
 #include "util/u_framebuffer.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_helpers.h"
 #include "util/u_math.h"
 #include "util/u_memory.h"
diff --git a/src/gallium/drivers/v3d/v3dx_emit.c b/src/gallium/drivers/v3d/v3dx_emit.c
index a8db6b85c6e..3f0675a5996 100644
--- a/src/gallium/drivers/v3d/v3dx_emit.c
+++ b/src/gallium/drivers/v3d/v3dx_emit.c
@@ -22,7 +22,7 @@
  */
 
 #include "util/format/u_format.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "v3d_context.h"
 #include "broadcom/common/v3d_macros.h"
 #include "broadcom/cle/v3dx_pack.h"
diff --git a/src/gallium/drivers/v3d/v3dx_state.c b/src/gallium/drivers/v3d/v3dx_state.c
index 4c03c90abe2..050bd3b81af 100644
--- a/src/gallium/drivers/v3d/v3dx_state.c
+++ b/src/gallium/drivers/v3d/v3dx_state.c
@@ -28,7 +28,7 @@
 #include "util/u_inlines.h"
 #include "util/u_math.h"
 #include "util/u_memory.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_helpers.h"
 #include "util/u_upload_mgr.h"
 
diff --git a/src/gallium/frontends/nine/vertexshader9.h b/src/gallium/frontends/nine/vertexshader9.h
index 0211decf80e..3c1b9ead1f9 100644
--- a/src/gallium/frontends/nine/vertexshader9.h
+++ b/src/gallium/frontends/nine/vertexshader9.h
@@ -23,7 +23,7 @@
 #ifndef _NINE_VERTEXSHADER9_H_
 #define _NINE_VERTEXSHADER9_H_
 
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 #include "iunknown.h"
 #include "device9.h"
diff --git a/src/gallium/tests/unit/translate_test.c b/src/gallium/tests/unit/translate_test.c
index a0c23d7a967..4d9c4e27ebf 100644
--- a/src/gallium/tests/unit/translate_test.c
+++ b/src/gallium/tests/unit/translate_test.c
@@ -27,7 +27,7 @@
 #include "translate/translate.h"
 #include "util/u_memory.h"
 #include "util/format/u_format.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_cpu_detect.h"
 #include "rtasm/rtasm_cpu.h"
 
diff --git a/src/gallium/tests/unit/u_half_test.c b/src/gallium/tests/unit/u_half_test.c
index 40b5ccbcaaa..7f2eba9382b 100644
--- a/src/gallium/tests/unit/u_half_test.c
+++ b/src/gallium/tests/unit/u_half_test.c
@@ -3,7 +3,7 @@
 #include <float.h>
 
 #include "util/u_math.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 #include "util/u_cpu_detect.h"
 
 static void
diff --git a/src/mesa/vbo/vbo_attrib_tmp.h b/src/mesa/vbo/vbo_attrib_tmp.h
index 81ef4b6a4c6..df61622670b 100644
--- a/src/mesa/vbo/vbo_attrib_tmp.h
+++ b/src/mesa/vbo/vbo_attrib_tmp.h
@@ -28,7 +28,7 @@ USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include "util/format_r11g11b10f.h"
 #include "main/varray.h"
 #include "vbo_util.h"
-#include "util/u_half.h"
+#include "util/half_float.h"
 
 
 /* ATTR */
diff --git a/src/util/format/u_format_pack.py b/src/util/format/u_format_pack.py
index 9cdd122d4fe..3545f9a0eae 100644
--- a/src/util/format/u_format_pack.py
+++ b/src/util/format/u_format_pack.py
@@ -715,7 +715,7 @@ def generate(formats):
     print()
     print('#include "pipe/p_compiler.h"')
     print('#include "util/u_math.h"')
-    print('#include "util/u_half.h"')
+    print('#include "util/half_float.h"')
     print('#include "u_format.h"')
     print('#include "u_format_other.h"')
     print('#include "util/format_srgb.h"')
diff --git a/src/util/half_float.c b/src/util/half_float.c
index 0d1e6a9551e..05aeac14abb 100644
--- a/src/util/half_float.c
+++ b/src/util/half_float.c
@@ -28,10 +28,10 @@
 #include <math.h>
 #include <assert.h>
 #include "half_float.h"
-#include "util/u_half.h"
 #include "rounding.h"
 #include "softfloat.h"
 #include "macros.h"
+#include "u_math.h"
 
 typedef union { float f; int32_t i; uint32_t u; } fi_type;
 
diff --git a/src/util/tests/format/u_format_test.c b/src/util/tests/format/u_format_test.c
index d33b41af8c2..f4a62a5c6a8 100644
--- a/src/util/tests/format/u_format_test.c
+++ b/src/util/tests/format/u_format_test.c
@@ -30,7 +30,8 @@
 #include <stdio.h>
 #include <float.h>
 
-#include "util/u_half.h"
+#include "util/half_float.h"
+#include "util/u_math.h"
 #include "util/format/u_format.h"
 #include "util/format/u_format_tests.h"
 #include "util/format/u_format_s3tc.h"
