diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index 9ad8e445b..9b0b830f5 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
+++ b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -67,11 +67,11 @@
 			compatible = "allwinner,simple-framebuffer",
 				     "simple-framebuffer";
 			allwinner,pipeline = "de_be0-lcd0-hdmi";
-			clocks = <&ccu CLK_AHB_LCD0>, <&ccu CLK_AHB_HDMI0>,
+			clocks = <&ccu CLK_AHB_HDMI0>,
 				 <&ccu CLK_AHB_DE_BE0>, <&ccu CLK_DE_BE0>,
 				 <&ccu CLK_TCON0_CH1>, <&ccu CLK_DRAM_DE_BE0>,
 				 <&ccu CLK_HDMI>;
-			status = "disabled";
+			status = "okay";
 		};
 
 		framebuffer-lcd0 {
@@ -92,7 +92,7 @@
 				 <&ccu CLK_AHB_DE_BE0>,
 				 <&ccu CLK_DE_BE0>, <&ccu CLK_TCON0_CH1>,
 				 <&ccu CLK_DRAM_TVE0>, <&ccu CLK_DRAM_DE_BE0>;
-			status = "disabled";
+			status = "okay";
 		};
 	};
 
@@ -649,6 +649,7 @@
 		};
 
 		hdmi: hdmi@1c16000 {
+			#sound-dai-cells = <0>; 
 			compatible = "allwinner,sun7i-a20-hdmi",
 				     "allwinner,sun5i-a10s-hdmi";
 			reg = <0x01c16000 0x1000>;
@@ -698,6 +699,7 @@
 			dmas = <&dma SUN4I_DMA_DEDICATED 29>,
 			       <&dma SUN4I_DMA_DEDICATED 28>;
 			dma-names = "rx", "tx";
+			allwinner,playback-channels = <8>;
 			status = "disabled";
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -947,6 +949,17 @@
 				       "PD15", "PD16", "PD17", "PD18", "PD19";
 				function = "lvds1";
 			};
+			
+			lcd0_rgb888_pins: lcd0-rgb888 {
+				pins = "PD0", "PD1", "PD2", "PD3",
+						"PD4", "PD5", "PD6", "PD7",
+						"PD8", "PD9", "PD10", "PD11",
+						"PD12", "PD13", "PD14", "PD15",
+						"PD16", "PD17", "PD18", "PD19",
+						"PD20", "PD21", "PD22", "PD23",
+						"PD24", "PD25", "PD26", "PD27";
+						function = "lcd0";
+			};
 
 			/omit-if-no-ref/
 			mmc0_pins: mmc0-pins {
@@ -1307,6 +1320,22 @@
 			       <&dma SUN4I_DMA_NORMAL 6>;
 			dma-names = "rx", "tx";
 			status = "disabled";
+		};
+		
+ 
+		hdmi_audio: hdmi_audio {
+			compatible = "simple-audio-card";
+			simple-audio-card,format = "i2s";
+			simple-audio-card,name = "allwinner-hdmi";
+			simple-audio-card,mclk-fs = <256>;
+			simple-audio-card,codec {
+				sound-dai = <&hdmi>;
+			};
+			simple-audio-card,cpu {
+				sound-dai = <&i2s0>;
+				dai-tdm-slot-num = <2>;
+				dai-tdm-slot-width = <32>;
+			};
 		};
 
 		rtp: rtp@1c25000 {

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index ee6900eb3..8b8d81e8e 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2499,27 +2499,102 @@
 	.connector_type = DRM_MODE_CONNECTOR_LVDS,
 };
 
-static const struct drm_display_mode lemaker_bl035_rgb_002_mode = {
-	.clock = 7000,
-	.hdisplay = 320,
-	.hsync_start = 320 + 20,
-	.hsync_end = 320 + 20 + 30,
-	.htotal = 320 + 20 + 30 + 38,
-	.vdisplay = 240,
-	.vsync_start = 240 + 4,
-	.vsync_end = 240 + 4 + 3,
-	.vtotal = 240 + 4 + 3 + 15,
+static const struct drm_display_mode lemaker_bl035_rgb_002_modes[] = {
+	{ /* 60 Hz */
+		.clock = 24000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 7,
+		.vsync_end = 480 + 7 + 6,
+		.vtotal = 480 + 7 + 6 + 7,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 50 Hz */
+		.clock = 24000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 40,
+		.vsync_end = 480 + 40 + 40,
+		.vtotal = 480 + 40 + 40 + 40,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 75 Hz */
+		.clock = 36000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 40,
+		.vsync_end = 480 + 40 + 40,
+		.vtotal = 480 + 40 + 40 + 40,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 100 Hz */
+		.clock = 48000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 40,
+		.vsync_end = 480 + 40 + 40,
+		.vtotal = 480 + 40 + 40 + 40,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 120 Hz */
+		.clock = 48000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 7,
+		.vsync_end = 480 + 7 + 6,
+		.vtotal = 480 + 7 + 6 + 7,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 30 Hz */
+		.clock = 12000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 7,
+		.vsync_end = 480 + 7 + 6,
+		.vtotal = 480 + 7 + 6 + 7,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
+	{ /* 25 Hz */
+		.clock = 12000,
+		.hdisplay = 640,
+		.hsync_start = 640 + 16,
+		.hsync_end = 640 + 16 + 30,
+		.htotal = 640 + 16 + 30 + 114,
+		.vdisplay = 480,
+		.vsync_start = 480 + 40,
+		.vsync_end = 480 + 40 + 40,
+		.vtotal = 480 + 40 + 40 + 40,
+		.flags = DRM_MODE_FLAG_NVSYNC | DRM_MODE_FLAG_NHSYNC,
+	},
 };
 
 static const struct panel_desc lemaker_bl035_rgb_002 = {
-	.modes = &lemaker_bl035_rgb_002_mode,
-	.num_modes = 1,
+	.modes = lemaker_bl035_rgb_002_modes,
+	.num_modes = ARRAY_SIZE(lemaker_bl035_rgb_002_modes),
+	.bpc = 8,
 	.size = {
 		.width = 70,
 		.height = 52,
 	},
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
-	.bus_flags = DRM_BUS_FLAG_DE_LOW,
 };
 
 static const struct drm_display_mode lg_lb070wv8_mode = {
